src/c/c_client.cpp

```diff
@@ -459,6 +459,52 @@ bool CompareCaseInsensitive(const char* a,const char* b) {
   return (*a == *b);
 }
 
+// Return True if the backend is TF or TFLITE
+bool _isTensorFlow(const char* backend)
+{
+  return CompareCaseInsensitive(backend, "TF") || CompareCaseInsensitive(backend, "TFLITE");
+}
+
+// Check the parameters common to all set_model functions
+void _check_params_set_model(void* c_client,
+                            const char* name, const char* backend, 
+                            const char** inputs, const size_t* input_lengths, const size_t n_inputs,
+                            const char** outputs, const size_t* output_lengths, const size_t n_outputs)
+{
+  // Sanity check params. Tag is strictly optional, and inputs/outputs are
+  // mandatory IFF backend is TensorFlow (TF or TFLITE)
+  SR_CHECK_PARAMS(c_client != NULL && name != NULL && backend != NULL);
+                  
+  if (_isTensorFlow(backend)) {
+    if (inputs == NULL || input_lengths == NULL ||
+        outputs == NULL || output_lengths == NULL) {
+      throw SRParameterException("Inputs and outputs are required with TensorFlow");
+    }
+  }
+
+  // For the inputs and outputs arrays, a single empty string is ok (this means
+  // that the array should be skipped) but if more than one entry is present, the
+  // strings must be nonzero length
+  if (_isTensorFlow(backend)) {
+    if (n_inputs != 1 && input_lengths[0] != 0) {
+      for (size_t i = 0; i < n_inputs; i++){
+        if (inputs[i] == NULL || input_lengths[i] == 0) {
+          throw SRParameterException(
+            std::string("inputs[") + std::to_string(i) + "] is NULL or empty");
```

Comment:
Because you're adding to a std::string (of `i`), I don't think you need to invoke the `std::string` constructor for "inputs["